gno.land/pkg/sdk/vm/keeper.go

```diff
@@ -440,9 +440,20 @@ func (vm *VMKeeper) AddPackage(ctx sdk.Context, msg MsgAddPackage) (err error) {
 	if !strings.HasPrefix(pkgPath, chainDomain+"/") {
 		return ErrInvalidPkgPath("invalid domain: " + pkgPath)
 	}
+
+	// Extra keeper-only checks.
+	gm, err := gnomod.ParseMemPackage(memPkg)
+	if err != nil {
+		return ErrInvalidPackage(err.Error())
+	}
+
 	if pv := gnostore.GetPackage(pkgPath, false); pv != nil {
-		return ErrPkgAlreadyExists("package already exists: " + pkgPath)
+		// Private packages can be re-uploaded (overwritten).
+		if !gm.Private {
```

Comment:
Careful: the `Private` property must be asserted on the package already present in the store, not the new one, otherwise it's a security breach, where any loaded package could be overwritten (by a private instance)!

Can you try again, this time testing `pv.Private`?

In this case, no need to compute `gm` before, as you did. The original order should still be correct.